From c3c7c35ccb37e5f0fee6fe72ec43da489365f058 Mon Sep 17 00:00:00 2001
From: Fabien Boucher <fboucher@redhat.com>
Date: Fri, 12 Mar 2021 16:25:55 +0100
Subject: [PATCH] Fix secret encoding issue to building the cookie

lodgeit[22609]: TypeError: string argument without an encoding
lodgeit[22609]: secret_key = bytes(secret_key)
lodgeit[22609]: File "/usr/lib/python3.6/site-packages/werkzeug/contrib/securecookie.py", line 155, in __init__
lodgeit[22609]: return cls(secret_key=secret_key)
lodgeit[22609]: File "/usr/lib/python3.6/site-packages/werkzeug/contrib/securecookie.py", line 302, in load_cookie
lodgeit[22609]: local.application.secret_key)
lodgeit[22609]: File "/usr/lib/python3.6/site-packages/lodgeit/utils.py", line 78, in __init__
lodgeit[22609]: request = Request(environ)
lodgeit[22609]: File "/usr/lib/python3.6/site-packages/lodgeit/application.py", line 56, in __call__
lodgeit[22609]: return self.app(environ, start_response)
lodgeit[22609]: File "/usr/lib/python3.6/site-packages/werkzeug/wsgi.py", line 600, in __call__
lodgeit[22609]: application_iter = app(environ, start_response)
lodgeit[22609]: File "/usr/lib/python3.6/site-packages/werkzeug/serving.py", line 197, in execute
lodgeit[22609]: execute(self.server.app)
lodgeit[22609]: File "/usr/lib/python3.6/site-packages/werkzeug/serving.py", line 209, in run_wsgi
lodgeit[22609]: Traceback (most recent call last):
---
 lodgeit/utils.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/lodgeit/utils.py b/lodgeit/utils.py
index 6ef3480..e745d30 100644
--- a/lodgeit/utils.py
+++ b/lodgeit/utils.py
@@ -75,7 +75,7 @@ class Request(RequestBase):
         super(Request, self).__init__(environ)
         self.first_visit = False
         session = SecureCookie.load_cookie(self, COOKIE_NAME,
-                                           local.application.secret_key)
+                                           local.application.secret_key.encode())
         user_hash = session.get('user_hash')
 
         if not user_hash:
-- 
2.26.2

